File: backend/app/models/language.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Language(Base):
    __tablename__ = "languages"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    name: Mapped[str] = mapped_column(String(100))
    locale: Mapped[str] = mapped_column(String(10), unique=True, index=True)
    icon_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    is_default: Mapped[bool] = mapped_column(Boolean)
    is_fallback: Mapped[bool] = mapped_column(Boolean)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    deleted_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True), nullable=True
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=datetime.utcnow
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow
    )

File: mobile/lib/models/language.dart
class Language {
  final String id;
  final String name;
  final String locale;
  final String? iconUrl;
  final bool isDefault;
  final bool isFallback;
  final bool isActive;

  const Language({
    required this.id,
    required this.name,
    required this.locale,
    this.iconUrl,
    this.isDefault = false,
    this.isFallback = false,
    this.isActive = true,
  });

  factory Language.fromJson(Map<String, dynamic> json) {
    return Language(
      id: json['id'] as String,
      name: json['name'] as String,
      locale: json['locale'] as String,
      iconUrl: json['icon_url'] as String?,
      isDefault: json['is_default'] as bool? ?? false,
      isFallback: json['is_fallback'] as bool? ?? false,
      isActive: json['is_active'] as bool? ?? true,
    );
  }
}
